clang/lib/Driver/Driver.cpp

```diff
@@ -3446,17 +3452,35 @@ class OffloadingActionBuilder final {
           if (!DeviceObjects.empty()) {
             // link and wrap the device binary, but do not perform the
             // backend compile.
+            // TODO: perform split here?
```

Comment:
I wasn't sure that we need split here. It seems like in this case there could be several device images with different types (aocx, aocr and llvm ir). And in case of aocx and aocr images split is not possible, so I'm not sure that we need to split remaining llvm ir modules. I hoped that someone on CR can give a right answer on this question.

Reply:
When aocx or aocr is seen, there is an expectation that none of the other objects processed have any spir kernel code.  These are effectively not processed by the BE compiler, but are still part of the final binary.

Reply:
Thanks, I'm seeing this now.
IMO, the split should not take place here for the reasons you've described, and therefore, neither should the SPIR-V Translation (so the next few lines need to be reverted).

@mdtoguchi, could you please sane-check and approve this point?

Reply:
The SPIR-V translation is needed - it was pulled out of the Link step.

Reply:
Thanks.
```suggestion
```

Reply:
I also added a clarifying comment, it's not always obvious what happens here.